### ISSUE TITLE
Do not raise unused-argument for singledispatch implementations when the argument is used in another implementation

### ISSUE THREAD

**Issue by anntzer** [PyCQA/pylint#1034]
### Steps to reproduce

In

```
from functools import singledispatch

@singledispatch
def func(arg):
    return arg

@func.register(int)
def _(arg):
    return 42

print(func(1), func("foo"))
```

pylint warns that the implementation of `func` for `int` does not use argument `arg`.  But similarly to #225, it typically makes sense to just maintain the same signature for all implementations.
### Current behavior

warning
### Expected behavior

no warning
### pylint --version output

pylint 1.6.4, 
astroid 1.4.7
Python 3.5.2 (default, Jun 28 2016, 08:46:01) 
[GCC 6.1.1 20160602]

**Comment by PCManticore** [comment]
This is fixed now in the master branch.
